src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -19,47 +19,7 @@ internal static class TypeExtensions
         public static MethodInfo GetAnyStaticMethodValidated(this Type type, string name, Type[] types)
         {
             Debug.Assert(types != null);
-            foreach (MethodInfo method in type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly))
-            {
-                if (method.Name == name && method.MatchesArgumentTypes(types))
-                {
-                    return method;
-                }
-            }
-
-            return null;
-        }
-
-        /// <summary>
-        /// Returns true if the method's parameter types are reference assignable from
-        /// the argument types, otherwise false.
-        ///
-        /// An example that can make the method return false is that
-        /// typeof(double).GetMethod("op_Equality", ..., new[] { typeof(double), typeof(int) })
-        /// returns a method with two double parameters, which doesn't match the provided
-        /// argument types.
-        /// </summary>
-        private static bool MatchesArgumentTypes(this MethodInfo mi, Type[] argTypes)
-        {
-            Debug.Assert(mi != null);
-            Debug.Assert(argTypes != null);
-
-            ParameterInfo[] ps = mi.GetParametersCached();
-
-            if (ps.Length != argTypes.Length)
-            {
-                return false;
-            }
-
-            for (int i = 0; i < ps.Length; i++)
-            {
-                if (!TypeUtils.AreReferenceAssignable(ps[i].ParameterType, argTypes[i]))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return type.GetMethod(name, BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static | BindingFlags.DeclaredOnly, null, types, null);
```

Comment:
This is not equivalent. In cases where there are two equally valid methods the existing code will return an arbitrary choice (indeed, it's happy to return one when there's an even better match) but this will throw `AmbiguousMatchException`.
I'm not sure that this isn't okay (it may even be an improvement) but I'd want to be more sure.